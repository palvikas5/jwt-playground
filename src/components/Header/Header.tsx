import React from 'react';
import NextLink from 'next/link';
import {
  StyledHeaderContainer,
  StyledHeader,
  HeaderTitle,
} from './header.style';

const Header = () => {
  return (
    <StyledHeaderContainer>
      <StyledHeader className='container'>
        <NextLink href='/' passHref>
          <HeaderTitle>JWT</HeaderTitle>
        </NextLink>
      </StyledHeader>
    </StyledHeaderContainer>
  );
};

export { Header };
