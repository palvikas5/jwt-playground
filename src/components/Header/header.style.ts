import styled from 'styled-components';

const StyledHeaderContainer = styled.div`
  color: white;
`;

const StyledHeader = styled.header`
  padding-top: 15px;
  padding-bottom: 15px;
`;

const HeaderTitle = styled.a`
  font-size: 2.5rem;
  display: inline-block;
  padding: 10px 0;
  color: inherit;
  text-decoration: none;
`;

export { StyledHeaderContainer, StyledHeader, HeaderTitle };
