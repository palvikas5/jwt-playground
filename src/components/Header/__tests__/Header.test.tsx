import React from 'react';
import { render } from '@testing-library/react';
import { Header } from '../Header';

describe('header', () => {
  it('should render successfully', () => {
    expect.assertions(1);

    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
});
