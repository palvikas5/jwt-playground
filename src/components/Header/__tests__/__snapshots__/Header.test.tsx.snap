// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header should render successfully 1`] = `
.c0 {
  color: white;
}

.c1 {
  padding-top: 15px;
  padding-bottom: 15px;
}

.c2 {
  font-size: 2.5rem;
  display: inline-block;
  padding: 10px 0;
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div>
  <div
    class="c0"
  >
    <header
      class="c1 container"
    >
      <a
        class="c2"
        href="/"
      >
        JWT
      </a>
    </header>
  </div>
</div>
`;
